### DISCUSSION TITLE
Incomplete feature.geometry during MapLayerMouseEvent

### DISCUSSION THREAD

**Discussion by lawrencenika** [maplibre/maplibre-gl-js#3922] [Q&A]
I have used Maplibre GL JS with Martin server to load geometries from PostGIS DB as a Filllayer. However, when I try to get the geometry using the following code, it appears to be cut off (possibly by the tile boundary), is there a way to get the complete geometry object?

```
    const handleDoubleClickForEdit = async (e: MapLayerMouseEvent) => {
        e.preventDefault(); // disable default of double click to zoom map in by 1 level
        if (!e.features || e.features.length < 1) return;

        const feature = e.features[0];
        const featureId = String(feature.id);
        const geom = feature.geometry; 
        const fc: FeatureCollection = geometryToFeatureCollection(geom, featureId);
        store.mapboxDraw.set(fc);
        ...
    }
```

-->

**maplibre-gl-js version**: 4.1.0

**browser**: Chrome

### Steps to Trigger Behavior

 1. Use my sample code

### Expected Behavior

<img width="633" alt="1" src="https://github.com/maplibre/maplibre-gl-js/assets/110647851/a1650943-995f-464f-8979-1e2dc926fa38">

### Actual Behavior

<img width="595" alt="2" src="https://github.com/maplibre/maplibre-gl-js/assets/110647851/cc4175b1-1745-4bf0-9e2e-8b607387b8f3">

**Comment by HarelM** [comment]
Geometries from tiles are cut in the source data and are not rebuilt in the client.
You can rebuild them, but it's not a picnic.
Also a jsbin would be helpful so that I won't be guessing what you tired to do.
If you need the full geometry I would advise to call the backend and get it.

**Reply by lawrencenika** [reply]
thank you. I will revert to fetch from DB directly. Is there a way to also modify a layer in maplibre directly given it uses martin function source to show geometries on the map ( so that the changes will be saved to DB async)?
